### PR TITLE
initial code

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: include/vtbx/global.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  #define VTBX_C_WARN_DISABLE_GCC(TEXT)
+#elif defined(VTBX_C_GNU)
+  #define VTBX_C_WARN_PUSH VTBX_C_PRAGMA(GCC diagnostic push)
+  #define VTBX_C_WARN_POP VTBX_C_PRAGMA(GCC diagnostic pop)
+  #define VTBX_C_WARN_DISABLE_CLANG(TEXT)
+  #define VTBX_C_WARN_DISABLE_GCC(TEXT) VTBX_C_PRAGMA(GCC diagnostic ignored TEXT)
+#endif // if defined(VTBX_C_CLANG)
+
+#define VTBX_C_WARN_DISABLE_LOGICAL_OP VTBX_C_WARN_DISABLE_GCC("-Wlogical-op")
+
```
Comment (benjamreis) [inline include/vtbx/global.h:66]: We should do the same for `clang`.

File: include/vtbx/global.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    #define VTBX_NO_DISCARD [[ nodiscard ]]
+  #else
+    #define VTBX_NO_DISCARD
+  #endif // if __cplusplus >= 201703L
+#else
+  #define VTBX_NO_DISCARD
+#endif // ifdef __cplusplus
+
+#define VTBX_UNUSED(ARG) ((void)ARG)
+
```
Comment (benjamreis) [inline include/vtbx/global.h:48]: Simpler:
```c
#if defined(__cplusplus) && __cplusplus >= 201703L
    #define VTBX_NO_DISCARD [[ nodiscard ]]
#else
    #define VTBX_NO_DISCARD
#endif // if defined(__cplusplus) && __cplusplus >= 201703L
```

File: src/io.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  } while (errno == EAGAIN || errno == EWOULDBLOCK || errno == EINTR);
+  VTBX_C_WARN_POP
+
+  return VTBX_ERR_ERRNO;
+}
+
+VtbxError vtbx_fd_wait_read (int fd, void *buf, size_t count, int timeout) {
+  if (timeout) {
+    VtbxError ret = vtbx_fd_wait_for_rdata(fd, timeout);
+    if (ret) return ret;
```
Comment (benjamreis) [inline src/io.c]: Add a comment: `// Here ret != VTBX_ERR_OK` to specify this is the error case.

File: include/vtbx/global.h (hunk 1 above)
Comment (benjamreis) [reply include/vtbx/global.h:66]: Useless, doesn't exist on clang.

File: include/vtbx/global.h (hunk 2 above)
Comment (Wescoeur) [reply include/vtbx/global.h:48]: :+1:

File: src/io.c (hunk 3 above)
Comment (Wescoeur) [reply src/io.c]: I think it's better to directly compare `ret` to `VTBX_ERR_OK`.

Comment (benjamreis) [reply src/io.c]: Even better indeed! :+1:

**Comment by benjamreis** [comment]
OK for merge.

**Pull request merged**